### PR TITLE
Add Request Processors pool documentation for WebApi

### PR DESCRIPTION
**Pull request by nuzil**
## This PR is a:

- [x] New topic
- [ ] Content update
- [ ] Content fix or rewrite
- [ ] Bug fix or improvement

## Summary

This documentation contains information about how Magento API are handling routing between Async / Bulk / Swagger and Usuall requests to Magento API

### CODE REVIEW

**Comment by magento-cicd2** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.3/extension-dev-guide/webapi/request-processor-pool.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+To create a custom processor, you must perform the following tasks: 
+
+* Define the custom processor in `webapi_rest/di.xml`.
+* Create a processor class and implement the `Magento\Webapi\Controller\Rest\RequestProcessorInterface` interface.
+* Define the processing rules in the `canProcess` method.
+* Create the processor logic in the `process` method.
+
+
+### Define the custom processors
```
Comment (erikmarr) [inline guides/v2.3/extension-dev-guide/webapi/request-processor-pool.md]: 'processors' should be 'processor'?

File: guides/v2.3/extension-dev-guide/webapi/request-processor-pool.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+     * @param \Magento\Framework\Webapi\Rest\Request $request
+     * @return void
+     * @throws \Magento\Framework\Exception\AuthorizationException
+     * @throws \Magento\Framework\Exception\InputException
+     * @throws \Magento\Framework\Webapi\Exception
+     */
+    public function process(\Magento\Framework\Webapi\Rest\Request $request);
+
+    /**
+     * Method should return true for all the request current processor can process.
```
Comment (erikmarr) [inline guides/v2.3/extension-dev-guide/webapi/request-processor-pool.md]: "...for all the requests the current processor can process."?

**Comment by keharper** [comment]
running tests

**Pull request merged**